webui/react/src/pages/ExperimentList.tsx

```diff
@@ -489,6 +489,29 @@ const ExperimentList: React.FC = () => {
     users,
   ]);
 
+  useEffect(() => {
+    // this is the 'failsafe' for columns settings
+    // if they get into a bad state
+    if (!settings.columns?.length || !settings.columnWidths?.length) {
+      updateSettings({
+        columns: DEFAULT_COLUMNS,
+        columnWidths: DEFAULT_COLUMNS.map((columnName) => DEFAULT_COLUMN_WIDTHS[columnName]),
+      });
+      return;
+    }
+    const columnNames = columns.map(column => column.dataIndex as ExperimentColumnName);
+    const actualColumns = settings.columns.filter((columnName) => columnNames.includes(columnName));
+    if (actualColumns.length < settings.columns.length) {
+      updateSettings({ columns: actualColumns });
+    }
+    if (settings.columnWidths.length !== actualColumns.length) {
+      updateSettings({
+        columnWidths:
+        actualColumns.map((columnName) => DEFAULT_COLUMN_WIDTHS[columnName]),
+      });
+    }
+  }, [ settings.columns, settings.columnWidths, columns, resetSettings, updateSettings ]);
+
```

Comment:
**suggestion:** slight update to reduce the number of consecutive `updateSettings` call:
```
  useEffect(() => {
    // This is the failsafe for when column settings get into a bad shape.
    if (!settings.columns?.length || !settings.columnWidths?.length) {
      updateSettings({
        columns: DEFAULT_COLUMNS,
        columnWidths: DEFAULT_COLUMNS.map((columnName) => DEFAULT_COLUMN_WIDTHS[columnName]),
      });
    } else {
      const columnNames = columns.map(column => column.dataIndex as ExperimentColumnName);
      const actualColumns = settings.columns.filter(name => columnNames.includes(name));
      const newSettings: Partial<ExperimentListSettings> = {};
      if (actualColumns.length < settings.columns.length) {
        newSettings.columns = actualColumns;
      }
      if (settings.columnWidths.length !== actualColumns.length) {
        newSettings.columnWidths = actualColumns.map(name => DEFAULT_COLUMN_WIDTHS[name]);
      }
      if (Object.keys(newSettings).length !== 0) updateSettings(newSettings);
    }
  }, [ settings.columns, settings.columnWidths, columns, resetSettings, updateSettings ]);
```

Reply:
incorporated!